### PR TITLE
Improve Documentation around JCasc and Custom Image

### PR DESCRIPTION
**Pull request by dduportal**
# What this PR does / why we need it

This PR fix some parts of the chart documentation to help newcomers around JCasC configuration:

- Add an explicit requirement about the mandatory plugins required by the chart
- Improve the JCasC configuration explanation for newcomers
- Fix the documented default value for the directive `controller.installPlugins`

# Which issue this PR fixes

- N/A.

# Special notes for your reviewer

🎅 Merry Christmas and Happy New Year!

# Checklist

- [x] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed

### CODE REVIEW

File: charts/jenkins/README.md [hunk 1]
```diff
@@ -112,7 +112,7 @@ controller:
   installPlugins: []
 ```
 
-Notice: `installPlugins` is set to an empty list to disable plugin download.
+Notice: `installPlugins` is set to an empty list to disable plugin download. In this case, the image `registry/my-jenkins:v1.2.3` must have the plugins specified as default value for [the `controller.installPlugins` directive](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/VALUES_SUMMARY.md#jenkins-plugins) to ensure that the configuration side-care system works as expected.
```
Comment (timja) [inline charts/jenkins/README.md:115]: (note beware of DCO check if applying suggestion from GitHub(

```suggestion
Notice: `installPlugins` is set to an empty list to disable plugin download. In this case, the image `registry/my-jenkins:v1.2.3` must have the plugins specified as default value for [the `controller.installPlugins` directive](https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/VALUES_SUMMARY.md#jenkins-plugins) to ensure that the configuration side-car system works as expected.
```

**Review by timja — DISMISSED** [review]

**Comment by timja** [comment]
>  Error validating maintainer 'timja': 502 Bad Gateway

flaked

**Comment by dduportal** [comment]
@timja thanks. Let me squash the latest commit to have a better history, and it should "un-flak" the build

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
weird issues, given that this is just docs changes, let's 🚢

**Pull request merged**